src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProject.cs

```diff
@@ -1064,21 +1174,30 @@ public void ProcessFileChange(string fullFilePath)
                             return;
                         }
 
+                        _nonSourceFilePathToSourceFileInfo.TryGetValue(fullFilePath, out var fileInfo);
+
                         _project._workspace.ApplyChangeToWorkspace(w =>
                         {
                             if (w.IsDocumentOpen(documentId))
                             {
                                 return;
                             }
 
-                            var documentInfo =
-                                DocumentInfo.Create(
-                                    document.Id,
-                                    document.Name,
-                                    document.Folders,
-                                    document.SourceCodeKind,
-                                    new FileTextLoader(fullFilePath, defaultEncoding: null),
-                                    document.FilePath);
+                            DocumentInfo documentInfo;
+                            if (fileInfo == null)
+                            {
+                                documentInfo = DocumentInfo.Create(
+                                      document.Id,
+                                      document.Name,
+                                      document.Folders,
+                                      document.SourceCodeKind,
+                                      new FileTextLoader(fullFilePath, defaultEncoding: null),
+                                      document.FilePath);
+                            }
+                            else
+                            {
+                                documentInfo = CreateDocumentInfoFromFileInfo(fileInfo, document.Folders);
```

Comment:
There's another blocking issue here. This document is already part of the workspace and we are processing a change. This method will return a `DocumentInfo` with a new `DocumentId` but we need to use `document.Id` (see line 1190). This will cause line 1202 to throw because the solution thinks it's not part of the solution.

Reply:
yep. my bad.